### ISSUE TITLE
[4.0] (3.9.x) Inconsistent behaviour in manage language overrides

### ISSUE THREAD

**Issue by coolcat-creations** [joomla/joomla-cms#28855]
### What needs to be fixed
In language overrides you have to first select context and language to find what you look for, it would be great if this section could work like all other components.

![grafik](https://user-images.githubusercontent.com/828371/80469764-54fd7e00-8941-11ea-9d70-12112af37a14.png)

### Why this should be fixed
more consistent workflow in backend

### How would you fix it
1) Context Dropdown
2) Language selection in Searchtools
3) possibility to change the language inside edit override view
4) maybe the possibility to translate/manage language overrides in the Multilanguage associations (?)

cc @infograf768

**Comment by infograf768** [comment]
> maybe the possibility to translate/manage language overrides in the Multilanguage associations (?)

That I would not be in favor as overrides are totally unrelated to associations and even multilingual sites (let's never forget that a site may propose multiple languages to let UI be defined on a user base but not be a multilingual site _per se_).

Concerning the present UI, it was a B/C improvement I did for J3.

Modifying it needs a total refactoring. See discussion here https://github.com/joomla/joomla-cms/pull/22107 and original issue stated there.
I welcome any proposal using AJAX.

**Comment by brianteeman** [comment]
If I have understood this correctly the reason for having to select the language/client first is because the "new" button needs to know the language and client and without filtering the overrides it doesnt know.

For me this can be resolved in the same way that creating a new article is resolved.

In the edit form you add the ability to select the language/client

So the very first field on the edit form is the select language/client which can be prefilled if the list had been filtered 

Am I missing something here. At least in theory this is a simple change

**Comment by coolcat-creations** [comment]
The best would be if it would be exactly the same usage as articles, I agree. Also a language override to be assigned to Al languages like articles and the identification which language overrides are missing.

**Comment by brianteeman** [comment]
>  the identification which language overrides are missing

Do you mean where a translator has not translated all the strings?

If so then I'm not sure how realistic that will be (at least for core) as the english string will often be in the language file so we cant tell if its not actually translated

**Comment by coolcat-creations** [comment]
I mean when I override "read more" to say "find out more" then it would be good for me as administrator to translate all the "read more" but often I miss a real overview in the override component. Maybe it could be even done in the Multilanguage Association Component?

**Comment by brianteeman** [comment]
oh I didnt realise that's what you meant. Definitely not a fan of that idea but if it scratches your itch ...

**Comment by coolcat-creations** [comment]
I guess it depends if you create a lot overrides or not. I can tell you that if you use a lot, it's absolute chaos which overrides exist, which not and what's maybe missing. Got a project with more than 800 Overrides and 9 languages and still stumble over strings that have been forgotten by the coworkers.

**Comment by brianteeman** [comment]
is this the same thing you resolved with https://github.com/joomla/joomla-cms/pull/46559

**Comment by coolcat-creations** [comment]
not quite, it would be good to have the language selection in the keys instead as a "preselector" in my opinion. It's still inconsistent regarding language selection in my opinion.
